### ISSUE TITLE
highlight_ignore_patterns is nil

### ISSUE THREAD

**Issue by PriceHiller** [levouh/tint.nvim#22]
Hello, enjoying the plugin thus far! I'm running into an issue with some of the latest commits, specifically 80b7b356 seems to be the culprit.

```
Error detected while processing VimEnter Autocommands for "*":
Error executing lua callback: ...site/pack/packer/start/tint.nvim/lua/tint/transforms.lua:14: bad argument #1 to 'ipairs' (table expected, got nil)
stack traceback:
        [C]: in function 'ipairs'
        ...site/pack/packer/start/tint.nvim/lua/tint/transforms.lua:14: in function 'transform_color'
        ...share/nvim/site/pack/packer/start/tint.nvim/lua/tint.lua:109: in function 'set_tint_ns'
        ...share/nvim/site/pack/packer/start/tint.nvim/lua/tint.lua:133: in function 'setup_namespaces'
        ...share/nvim/site/pack/packer/start/tint.nvim/lua/tint.lua:353: in function <...share/nvim/site/pack/packer/start/tint.nvim/lua/tint.lua:348>
```

When checking it out, it appears that the addition of permitting a skip of [`setup_user_config`](https://github.com/levouh/tint.nvim/blob/2b7265cd180ac9be0cd0000e2abf48978930a1bf/lua/tint.lua#L349) results in the `user_config` lacking transforms leading to a `nil` value being passed to `ipairs` causing the error above.

For me, a fix that works currently is to remove the ability to skip that `setup_user_config`, but it seems that may not be optimal given the functionality was added in the first place.

**Comment by levouh** [comment]
Will take a look, apologies for the issues.

**Comment by levouh** [comment]
@treatybreaker what is your current config setup? Mind pasting it here?

**Comment by PriceHiller** [comment]
First off: thanks for the quick response! No need to apologize, you're providing this plugin for _free_ and of your own time given. I appreciate that you created it. 😄

Secondly, here's my configuration (via packer):

```lua
use({
    "levouh/tint.nvim",
    config = function()
        require("tint").setup({
            highlight_ignore_patterns = { "WinSeparator" },
            tint = -30,
            saturation = 0.8,
        })
    end,
})
```

**Comment by levouh** [comment]
Ahh I see the issue - the function (`__.setup_all`) gets called from an `:h autocmd` which in Lua will have the first argument as a table of data, hence `skip_config` always being `true`.

**Comment by levouh** [comment]
Should be fixed by #23, let me know if not. Cheers for the report and investigation around what caused the issue.

**Comment by PriceHiller** [comment]
Hey I appreciate the quick!

I was looking at it and I'd like to bounce this idea off of you instead of the current fix implemented (not that there's anything wrong with #23, but I might have a smaller solution):

In the `VimEnter` autocmds we can just wrap the `func` callback up in another function and call it like this:

```lua
--- Ensure the passed function runs after `:h VimEnter` has run
---
---@param func function The function to call only after `VimEnter` is done
local function on_or_after_vimenter(func)
  if vim.v.vim_did_enter == 1 then
    func()
  else
    vim.api.nvim_create_autocmd({ "VimEnter" }, {
      callback = function()
        func()
      end,
      once = true,
    })
  end
end

```
This resolves the issue as well.

I literally just considered that not even 30 seconds before you opened that PR.

Just a thought I had, probably missing some functionality I hadn't considered.

**Comment by PriceHiller** [comment]
Take a look at [this branch](https://github.com/treatybreaker/tint.nvim/tree/fix-autocmd-callbacks) if you're interested at all. If not, then no problems. 

Again thanks for the quick resolution, plugin is pretty hype.
